src/client/features/textDocument.ts

```diff
@@ -124,7 +124,10 @@ export class TextDocumentDidOpenFeature extends TextDocumentNotificationFeature<
         if (documentSelector && textDocumentSyncOptions && textDocumentSyncOptions.openClose) {
             this.register(this.messages, {
                 id: uuidv4(),
-                registerOptions: { documentSelector },
+                registerOptions: {
+                    documentSelector,
+                    providerName: this.client.name,
```

Comment:
This is mixing the concepts of a client name (which is always set to the extension ID) and the provider name (there can be multiple providers per extension). This is fine for simplicity for now, but if you choose to keep these conflated, document that it is a temporary simplification.

Reply:
Since the client ID is [synonymous with extension ID](https://sourcegraph.sgdev.org/github.com/sourcegraph/sourcegraph-extension-api@master/-/blob/src/client/client.ts#L125:40&tab=references), and the client name was [completely unused](https://sourcegraph.sgdev.org/github.com/sourcegraph/sourcegraph-extension-api@master/-/blob/src/client/client.ts#L125:68&tab=references) prior to this PR, maybe these concepts (client ID, client name, extension ID, provider name) could be unified under extension ID, which means we don't need this new provider name concept 🤔 

@sqs I don't see multiple providers per extension causing ambiguity because the client always asks the controller for results for one capability at a time (e.g. `getHover`) and each extension either provides or does not provide the capability - that is, no extension will provide a capability more than once. Please correct me if I'm not understanding a relevant part of this 🙇 

More concretely, we could:

- Get rid of client name (unused)
- Do not introduce provider name in this PR
- Unify client ID and extension ID (just call it extension ID)
- Change the return type of methods such as `getHover` from `HoverMerged` to `{ extension: ID, response: Hover }[]` (essentially tagging responses with the extension ID from which they came, and this could be factored out as a type function for convenience), and transfer responsibility for merging results to the clients such as the web app (this is a half baked idea, might not work)

Reply:
I ended up removing `Client.name`, since it was identical to `Client.id` in all instances.